Shoko.Server/Services/AbstractUserDataService.cs

```diff
@@ -235,6 +240,80 @@ public async Task<bool> SetEpisodeWatchedStatus(IShokoUser user, IShokoEpisode e
 
     #endregion
 
+    #region Series User Data
+
+    public async Task VoteOnSeries(IShokoSeries series, decimal voteValue, VoteType voteType, IShokoUser? user = null)
+    {
+        ArgumentNullException.ThrowIfNull(series);
+
+        if (user == null)
+            user = RepoFactory.JMMUser.GetAll().FirstOrDefault(u => u.IsAdmin == 1);
+
+        ArgumentNullException.ThrowIfNull(user);
+
+        if (series is not SVR_AnimeSeries svrSeries)
+            throw new ArgumentException("Series must be a SVR_AnimeSeries", nameof(series));
+
+        if (svrSeries.AniDB_Anime is not { } anidbAnime)
+            throw new ArgumentException("AniDB anime is not available for the series! Aborting!");
+
+        if (voteValue < -1 || voteValue > 10)
```

Comment:
since `voteValue` is a decimal then we should probably check if it is exactly `-1` or between `0` and `10`, so it cannot be -0.00001 and make havoc on the logic below.